modules/net-vpc-peering/main.tf

```diff
@@ -21,7 +21,7 @@ locals {
 }
 
 resource "google_compute_network_peering" "local_network_peering" {
-  name         = "${local.prefix}${local.local_network_name}-${local.peer_network_name}"
+  name         = var.override_name != null ? var.override_name : "${local.prefix}${local.local_network_name}-${local.peer_network_name}"
```

Comment:
this will end up being something like
```terraform
  name = coalesce(
    var.name.local,
    "${local.prefix}${local.local_network_name}-${local.peer_network_name}"
  )
```
As a plus, you could add a [precondition](https://developer.hashicorp.com/terraform/language/expressions/custom-conditions#preconditions-and-postconditions) to ensure the user provides a name if the default is longer than the maximum allowed lenght.